### PR TITLE
Update Github Issue & PR Templates

### LINKED ISSUE #489: [Suggestion] update Github Issue & PR Template
I've noticed that the project has an outdated Github Issue & PR Template format. As displayed by Github warning asking to update it.

<details>
<summary>Github Warning</summary>

image

</details>

I think it would be better to update the Github Issue & PR template, plus add other missing templates.

I will be opening an PR soon about this, let me known what you guys think.

### PR DESCRIPTION
**Pull request by santosned**
# Description

As the title suggest, this PR will updates the Github Issue & PR Template.

## Changes

- Remove old ISSUE_TEMPLATE.md
- Add new ISSUE_TEMPLATE folder as recommended by Github
- Add Github issue template config.yml
- Add Github issue template for bug reports
- Add Github issue template for feature requests
- Add Github issue template for documentation fixes
- Add Github issue template for documentation new
- Add Github simple initial PR template
- Add Github PR template for bug fixes

## Additional context

Closes #489

### CODE REVIEW

**Review by Shougo — CHANGES_REQUESTED** [review]
Please don't remove `minimal init.vim config` for `bug report` and `bug fixes`.
It is required.

**Comment by santosned** [comment]
I'll make a commit updating them soon.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+### Minimal Config
+
+⚠️ Required when reporting bugs about plugins or the plugin manager.
+
+See a minimal config example at [README#Config Example](https://github.com/Shougo/dein.vim#config-example), you can remove comments unrelated to the issue.
+
+```vim
+" Add your minimal .vimrc or init.vim here.
+
```
Comment (Shougo) [inline .github/ISSUE_TEMPLATE/bug_report.md:54]:
```suggestion
" Your minimal .vimrc
set runtimepath^=~/path/to/dein.nvim/
call dein#begin(path)
call dein#end()
```

File: .github/PULL_REQUEST_TEMPLATE/bug_fixes.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+### Minimal Config
+
+⚠️ Required when fixing bugs about plugins or the plugin manager.
+
+See a minimal config example at [README#Config Example](https://github.com/Shougo/dein.vim#config-example), you can remove comments unrelated to the pull request.
+
+```vim
+" Add your minimal .vimrc or init.vim here.
+
```
Comment (Shougo) [inline .github/PULL_REQUEST_TEMPLATE/bug_fixes.md:22]:
```suggestion
" Add your minimal .vimrc or init.vim here.
set runtimepath^=~/path/to/dein.nvim/
call dein#begin(path)
call dein#end()
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+ - OS: [e.g. iOS]
+ - Vim or Neovim version: [e.g. Vim v9]
+ - Commit/Version: [e.g. full SHA1 signature or link to the commit]
+
+### Minimal Config
+
+⚠️ Required when reporting bugs about plugins or the plugin manager.
+
+See a minimal config example at [README#Config Example](https://github.com/Shougo/dein.vim#config-example), you can remove comments unrelated to the issue.
```
Comment (Shougo) [inline .github/ISSUE_TEMPLATE/bug_report.md:50]:
```suggestion
See the minimal config example at [README#Config Example](https://github.com/Shougo/dein.vim#config-example), you can remove comments unrelated to the issue.
```

File: .github/PULL_REQUEST_TEMPLATE/bug_fixes.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+## Bug Fixes
+
+A clear and concise description of what the pull request does.
+
+### Minimal Config
+
+⚠️ Required when fixing bugs about plugins or the plugin manager.
+
+See a minimal config example at [README#Config Example](https://github.com/Shougo/dein.vim#config-example), you can remove comments unrelated to the pull request.
```
Comment (Shougo) [inline .github/PULL_REQUEST_TEMPLATE/bug_fixes.md:18]:
```suggestion
See the minimal config example at [README#Config Example](https://github.com/Shougo/dein.vim#config-example), you can remove comments unrelated to the pull request.
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+about: Create a report to help us improve
+title: ''
+labels: ''
+assignees: ''
+
+---
+
+## Related Issues
+
+Are there any existing issues or PRs that relate to this problem? If so, link them here.
```
Comment (Shougo) [inline .github/ISSUE_TEMPLATE/bug_report.md:12]:
```suggestion
Are there any existing issues or PRs that related to this problem? If so, link them here.
```

**Comment by Shougo** [comment]
OK.  Merged.

**Pull request merged**